HeartbeatLogging/Sources/HeartbeatsPayload.swift

```diff
@@ -84,21 +84,20 @@ extension HeartbeatsPayload: HTTPHeaderRepresentable {
   /// Returns a processed payload string intended for use in a HTTP header.
   /// - Returns: A string value from the heartbeats payload.
   public func headerValue() -> String {
-    // TODO(ncooke3): Return a empty payload instead of empty string.
-    if userAgentPayloads.isEmpty {
-      return ""
-    }
-
     let encoder = JSONEncoder()
     encoder.dateEncodingStrategy = .formatted(Self.dateFormatter)
 
-    // TODO(ncooke3): Fall back to base64URL-only if gzipping fails.
+    guard let data = try? encoder.encode(self) else {
+      // Because `HeartbeatsPayload` conforms to Codal
+      return Self.emptyPayload.headerValue()
+    }
+
     do {
-      let data = try encoder.encode(self)
       let gzippedData = try data.zipped()
       return gzippedData.base64URLEncodedString()
     } catch {
-      return "" // Return empty string if processing failed.
+      // If gzipping fails, fall back to encoding with base64URL.
+      return data.base64URLEncodedString()
```

Comment:
2. Previously, if gzipping fails, we would return an empty string. In practice, I'm not sure when gzip would fail, but if it did, we could probably just fall back to encoding the data without zipping to prevent losing the heartbeat data. From the payload analysis doc I wrote, it would be unlikely that an unzipped payload would be too large.